packages/react/src/components/ThemeProvider/index.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import { createTheme, Theme } from '@aws-amplify/ui';
+
+import { ThemeContext, ColorMode } from './ThemeContext';
+
+interface ThemeProviderProps {
+  children: React.ReactNode;
+  colorMode?: ColorMode;
+  theme?: Theme;
+  root?: boolean;
+}
+
+export { ThemeContext, ColorMode };
+
+export function ThemeProvider({
+  children,
+  colorMode,
+  theme,
+  root,
+}: ThemeProviderProps) {
+  // createTheme will always return a fully formed theme object,
```

Comment:
Should a modified version of this comment be a JSDoc comment (shown in intellisense) on the ThemeProviderProps.theme property so that customers know that their theme will extend from default theme?

```
interface ThemeProviderProps {
  children: React.ReactNode;
  colorMode?: ColorMode;
  /** 
   * Custom theme extended from the default theme.
  */
  theme?: Theme;
  root?: boolean;
}
```

On that note, shouldn't the other properties have jsdoc comments so customers know what they are for?